diff -Naru ghostscript-10.06.0_orig/brotli/c/dec/decode.c ghostscript-10.06.0/brotli/c/dec/decode.c
--- ghostscript-10.06.0_orig/brotli/c/dec/decode.c	2025-09-09 18:36:39.000000000 +0900
+++ ghostscript-10.06.0/brotli/c/dec/decode.c	2026-03-02 15:59:09.538828776 +0900
@@ -434,11 +434,11 @@
                                         BrotliBitReader* br,
                                         brotli_reg_t* bits,
                                         brotli_reg_t* value) {
+  BROTLI_HC_MARK_TABLE_FOR_FAST_LOAD(table);
+  BROTLI_HC_ADJUST_TABLE_INDEX(table, BrotliGetBits(br, HUFFMAN_TABLE_BITS));
   if (safe) {
     return;
   }
-  BROTLI_HC_MARK_TABLE_FOR_FAST_LOAD(table);
-  BROTLI_HC_ADJUST_TABLE_INDEX(table, BrotliGetBits(br, HUFFMAN_TABLE_BITS));
   *bits = BROTLI_HC_FAST_LOAD_BITS(table);
   *value = BROTLI_HC_FAST_LOAD_VALUE(table);
 }
