diff -Naru ghostscript-10.08.0_orig/pdf/pdf_shading.c ghostscript-10.08.0/pdf/pdf_shading.c
--- ghostscript-10.08.0_orig/pdf/pdf_shading.c	2026-09-08 17:42:15.000000000 +0900
+++ ghostscript-10.08.0/pdf/pdf_shading.c	2026-09-09 07:19:48.993687780 +0900
@@ -53,7 +53,7 @@
     if (code < 0)
         return code;
 
-    code = pdfi_dict_get_int(ctx, shading_dict, "ShadingType", &type);
+    code = pdfi_dict_get_int(ctx, shading_dict, "ShadingType", (int64_t *)(&type));
     if (code < 0)
         goto build_shading_function_error;
 
