diff -Naru cups-2.4.16_orig/cups-config.in cups-2.4.16/cups-config.in
--- cups-2.4.16_orig/cups-config.in	2025-12-04 17:13:12.000000000 +0900
+++ cups-2.4.16/cups-config.in	2025-12-05 18:05:35.216838879 +0900
@@ -44,7 +44,7 @@
 	CFLAGS="$CFLAGS -I$includedir"
     fi
 
-    if test $libdir != /usr/lib -a $libdir != /usr/lib32 -a $libdir != /usr/lib64; then
+    if test $libdir != /usr/lib -a $libdir != /usr/lib -a $libdir != /usr/lib; then
 	LDFLAGS="$LDFLAGS -L$libdir"
     fi
 fi
