diff -Naru cups-2.3.3_orig/ppdc/Makefile cups-2.3.3/ppdc/Makefile
--- cups-2.3.3_orig/ppdc/Makefile	2020-04-28 03:04:29.000000000 +0900
+++ cups-2.3.3/ppdc/Makefile	2025-12-04 12:33:47.599568520 +0900
@@ -182,12 +182,7 @@
 
 genstrings:		genstrings.o libcupsppdc.a ../cups/$(LIBCUPSSTATIC) \
 			sample.drv ../data/media.defs
-	echo Linking $@...
-	$(LD_CXX) $(ARCHFLAGS) $(ALL_LDFLAGS) -o genstrings genstrings.o \
-		libcupsppdc.a $(LINKCUPSSTATIC)
-	$(CODE_SIGN) -s "$(CODE_SIGN_IDENTITY)" $@
-	echo Generating localization strings...
-	./genstrings >sample.c
+	echo Not Linking $@...
 
 
 #
